search/includes/classes/class-queue.php

```diff
@@ -737,7 +765,7 @@ private function delete_jobs_on_the_same_object( $all_deadlocked_jobs ) {
 	/**
 	 * We can't re-queue jobs that are already waiting in queue. We should remove such jobs instead.
 	 */
-	private function delete_jobs_on_the_already_queued_object( $deadlocked_jobs ) {
+	public function delete_jobs_on_the_already_queued_object( $deadlocked_jobs ) {
```

Comment:
Why are we making this function public?

Reply:
One of the tests (`test_free_deadlocked_jobs_handle_duplicates`) is mocking this function. It was throwing warnings before and kinda pass by mistake. So I made it public so it is easily mockable. I know not the best approach but as we dont really use inversion of controll it is hard to mock the dependencies that this function calls. It is way easier just to mock the function.

Reply:
If it's just for tests, we can use something similar to what I did here: https://github.com/Automattic/vip-go-mu-plugins/blob/14f6d02bb262fbbbe4952a12790d5caa2d525343/tests/vip-jetpack/connection-pilot/test-class-jetpack-connection-pilot.php#L8

Reply:
@pauarge that's neat!

Reply:
Hmm, interesting trick:) I don't think it would work in this case though.

The test calls another public method, which in turn calls this previously private method. The way you suggested you operate on another ReflectionClass, so whatever config/mocking we would do there would not be used by the production code that calls this private methed..